crates/ruff/src/main.rs

```diff
@@ -87,14 +87,19 @@ pub fn main() -> ExitCode {
         Err(err) => {
             #[allow(clippy::print_stderr)]
             {
+                use std::io::Write;
+
+                // Use `writeln` instead of `eprintln` to avoid panicking when the stderr pipe is broken.
+                let mut stderr = std::io::stderr().lock();
+
                 // This communicates that this isn't a linter error but ruff itself hard-errored for
                 // some reason (e.g. failed to resolve the configuration)
-                eprintln!("{}", "ruff failed".red().bold());
+                writeln!(stderr, "{}", "ruff failed".red().bold()).ok();
```

Comment:
Looking at the original trace, I strongly suspect that ruff panics here... when trying to print the reason why ruff errored:

```rust
failed printing to stderr: Broken pipe (os error 32)
   0: ruff_server::message::init_messenger::{{closure}}
   1: std::panicking::rust_panic_with_hook
   2: std::panicking::begin_panic_handler::{{closure}}
   3: std::sys_common::backtrace::__rust_end_short_backtrace
   4: rust_begin_unwind
   5: core::panicking::panic_fmt
   6: std::io::stdio::_eprint
   7: ruff::main
   8: std::sys_common::backtrace::__rust_begin_short_backtrace
   9: std::rt::lang_start::{{closure}}
  10: std::rt::lang_start_internal
  11: main
  12: <unknown>
  13: __libc_start_main
  14: <unknown>
```